rdmo/views/signals.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+from django.contrib.auth.models import User, Group
+from django.contrib.sites.models import Site
+
+from rdmo.questions.models import Catalog
+from rdmo.projects.models import Project, Membership
+
+from .models import View
+
+logger = logging.getLogger(__name__)
+
+def m2m_changed_view_catalog_signal(sender, instance, **kwargs):
+    catalogs = instance.catalogs.all()
+
+    if catalogs.count() > 0:
```

Comment:
Her I would use `catalogs` since the `all()` will already hit the database.

Reply:
When using this suggestion in combination with the list comprehension, the following error will be raised:
> 'ManyRelatedManager' object is not iterable

So I guess at some point we have to call the db or is there an alternative to using `all()`?

Reply:
I mean

```
catalogs = instance.catalogs.all()

if catalogs:
    catalog_candidates = Catalog.objects.exclude(id__in=[catalog.id for catalog in catalogs])
```

That should work, you are right, we need to call the db at one point with `all()` or with `catalogs.values_list('pk')`. But we don't need both.

Reply:
Oh, sorry. I looked at the wrong line. Will adjust the code accordingly.

Reply:
I removed the `count()` calls.